torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -980,17 +994,138 @@ const Expr* PolynomialTransformer::mutate(const Max* v) {
     return evaluateOp(new Max(lhs_new, rhs_new, v->propagate_nans()));
   }
 
+  // If diff is constant, return the appropriate operand.
   const Expr* diff = new Sub(lhs_new, rhs_new);
   diff = diff->accept_mutator(this);
-  if (!diff->isConstant()) {
+  if (diff->isConstant()) {
+    if (immediateAs<int>(diff) > 0) {
+      return lhs_new;
+    }
+    return rhs_new;
+  }
+
+  auto combine_scalars = [&](const Expr* c1, const Expr* c2) -> const Expr* {
```

Comment:
I tried doing that, but that introduces complications while finally generating code from `MinTerm` / `MaxTerm`. This is because, if the `scalar` is set to the minimum value at that point, that has to be ignored. I think keep it null is simpler.

Let me know if you strongly believe this should not be done.